File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.java
class Solution {
    public int solution(int[] stones, int k) {
        int answer = 0;
        
        int left = 0, right = 200000001;
        
        while(left <= right){
            int mid = (left + right) / 2;
            int cnt = 0;
            boolean flag = false;
            
            for(int i=0 ; i<stones.length ; i++){
                if(stones[i] - mid <= 0)cnt++;
                else cnt = 0;
                if(cnt >= k){
                    flag = true;
                    break;
                }
            }
            
            if(flag){
                right = mid - 1;
            }else{
                left = mid + 1;
                answer = Math.max(answer, mid);
            }
        }
        
        return answer + 1;
    }
}

File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.py
def solution(stones, k):
    answer = 0
    
    left, right = 0, 2 * int(1e8)
    
    while left <= right:
        mid = (left + right) // 2
        cnt = 0
        can = True
        
        for stone in stones:
            if stone < mid:
                cnt += 1
            else:
                cnt = 0
                
            if cnt == k:
                can = False
                break
                
        if can:
            left = mid + 1
            answer = max(answer, mid)
        else:
            right = mid - 1
    return answer